### PR TITLE
WIP: use existing generated certs to map into self-hosted etcd expects

### PR DESCRIPTION
**Pull request by aaronlevy**
This is just a straw-man do not merge.

### CODE REVIEW

File: pkg/asset/asset.go [hunk 1]
```diff
@@ -25,15 +25,8 @@ const (
 	AssetPathEtcdClientKey                  = "tls/etcd-client.key"
 	AssetPathEtcdPeerCert                   = "tls/etcd-peer.crt"
 	AssetPathEtcdPeerKey                    = "tls/etcd-peer.key"
-	AssetPathSelfHostedOperatorEtcdCA       = "tls/operator/etcd-ca-crt.pem"
```
Comment (hongchaodeng) [inline pkg/asset/asset.go]: This is needed for self-hosted etcd path.

File: pkg/util/etcdutil/migrate.go [hunk 2]
```diff
@@ -237,7 +237,7 @@ func cleanupBootstrapEtcdService(kubecli kubernetes.Interface) {
 }
 
 func detectEtcdTLS(assetDir string) (bool, error) {
-	etcdCAAssetPath := filepath.Join(assetDir, asset.AssetPathSelfHostedOperatorEtcdCA)
+	etcdCAAssetPath := filepath.Join(assetDir, asset.AssetPathEtcdCA)
```
Comment (hongchaodeng) [inline pkg/util/etcdutil/migrate.go]: This won't work for recovery path.

**Comment by hongchaodeng** [comment]
Most of the changes here looks good.
But please test recovery path to make sure it works before merging the PR.

**Comment by aaronlevy** [comment]
I'll wait for the naming changes in etcd-operator before picking this up again

**Comment by aaronlevy** [comment]
closing this as the changes will be in: https://github.com/kubernetes-incubator/bootkube/pull/615

**Pull request closed without merging**